Cilium crashes when syncing metrics map

@rcorrear on Slack reports:

Cilium version: 1.1.0-rc4 678eda332

```
level=info msg="Launching Cilium health endpoint"
level=info msg="Enabling docker event listener" subsys=workload-watcher
panic: runtime error: index out of range

goroutine 236 [running]:
github.com/cilium/cilium/pkg/maps/metricsmap.SyncMetricsMap(0x0, 0x0)
        /go/src/github.com/cilium/cilium/pkg/maps/metricsmap/metricsmap.go:194 +0x33b
created by github.com/cilium/cilium/pkg/controller.(*Manager).UpdateController
        /go/src/github.com/cilium/cilium/pkg/controller/manager.go:82 +0x30c
```

Code:

https://github.com/cilium/cilium/blob/e542412662b85a1b32bb80043d115e03c9b3b83f/pkg/maps/metricsmap/metricsmap.go#L190-L195


Workaround: Moved back to `stable` version.

Marking this as a potential 1.1 release blocker as it causes an unrecoverable crash for users, depending on their CPU configuration.

--- a/pkg/maps/metricsmap/metricsmap.go
+++ b/pkg/maps/metricsmap/metricsmap.go
@@ -167,7 +167,7 @@ func updatePrometheusMetrics(key *Key, val *Value) {
 // aggregating it into drops (by drop reason and direction) and
 // forwards (by direction) with the prometheus server.
 func SyncMetricsMap() error {
-	var entry [possibleCPUsFileLength]Value
+	entry := make([]Value, possibleCpus)
 	file := bpf.MapPath(MapName)
 	metricsmap, err := bpf.OpenMap(file)
 
@@ -182,7 +182,7 @@ func SyncMetricsMap() error {
 		if err != nil {
 			break
 		}
-		err = bpf.LookupElement(metricsmap.GetFd(), unsafe.Pointer(&nextKey), unsafe.Pointer(&entry))
+		err = bpf.LookupElement(metricsmap.GetFd(), unsafe.Pointer(&nextKey), unsafe.Pointer(&entry[0]))
 		if err != nil {
 			return fmt.Errorf("unable to lookup metrics map: %s", err)
 		}

